src/fiber/tests/payment.rs

```diff
@@ -940,58 +741,506 @@ async fn test_send_payment_two_nodes_with_hop_hints_and_multiple_channels() {
             max_parts: None,
             keysend: Some(true),
             udt_type_script: None,
-            allow_self_payment: true,
+            allow_self_payment: false,
             hop_hints: Some(vec![
-                // node1 - channel_1 -> node2
                 HopHint {
-                    pubkey: node_0.pubkey.clone(),
-                    channel_funding_tx: channel_1_funding_tx,
-                    inbound: false,
+                    pubkey: node2.pubkey.clone(),
+                    channel_outpoint: outpoint1,
+                    fee_rate: DEFAULT_TLC_FEE_PROPORTIONAL_MILLIONTHS as u64,
+                    tlc_expiry_delta: DEFAULT_TLC_EXPIRY_DELTA,
                 },
-                // node2 - channel_3 -> node1
                 HopHint {
-                    pubkey: node_0.pubkey.clone(),
-                    channel_funding_tx: channel_3_funding_tx,
-                    inbound: true,
+                    pubkey: node2.pubkey.clone(),
+                    channel_outpoint: outpoint2,
+                    fee_rate: DEFAULT_TLC_FEE_PROPORTIONAL_MILLIONTHS as u64,
+                    tlc_expiry_delta: DEFAULT_TLC_EXPIRY_DELTA,
                 },
             ]),
             dry_run: false,
         })
-        .await
-        .unwrap();
+        .await;
 
+    assert!(res.is_ok(), "Send payment failed: {:?}", res);
```

Comment:
from the code in find_path, seems we will first try to pick channels with hint, should we add unit test for channels with and without hint, and make sure channels with hint will be selected?

Reply:
As discussion on discord, the decision is that
> 那就还是简单点？ hints 的意思就是 hints，真正用的 payment path 可能就是不采用这些 hints。用户想要限制使用某些路径就用 send_to_route